### DISCUSSION TITLE
Map a key sequence to type a character

### DISCUSSION THREAD

**Discussion by RonSMeyer** [sezanzeb/input-remapper#119] [Q&A]
I want to map a key sequence to type a character. xmodmap says ";" and ":" are keycode-47. I want to map keycode-47 so that it still types ";" and Shift-keycode47 types ":", but additionally Alt-L+keycode47 types the character "ö" and Cntl-L+keycode47 types the character "Ö". The concept seems simple, "Alt-L+this-key types this-value". How do you do that?

**Comment by sezanzeb** [comment]
with macros: `m(Shift_L, h(comma))` writes `;` on my computer.

Depending on what 47 is on your keyboard, for me it's `,`

**Reply by RonSMeyer** [reply]
On the en_US keyboard, keycode 47 is the semi-colon,colon key immediately to the right of the letter 'L'.  I want to map that key specifically (along with others) because it is the same location as the "ö" character on a German keyboard. I am taking an online class and copying these special characters from Character Map is getting to be a real pain. The actual key itself does not matter. The question is mapping "modifier + some-key" to type a designated character on the screen.  

In the software field **KEY** I typed the sequence, and it shows **"Alt L + semicolon"**.
In the **MAPPING** field I put k(ö). It says "Unknown key".  So.......how exactly do you do these macros?

**Reply by sezanzeb** [reply]
you just insert the macro code as a mapping as shown in the screenshot: https://raw.githubusercontent.com/sezanzeb/key-mapper/main/readme/screenshot.png

**Reply by RonSMeyer** [reply]
Trying a third time. With the key sequence "Alt_L + semicolon" I want it to type the character ö .  It says unknown key. 

![Screenshot from 2021-06-06 10-59-14](https://user-images.githubusercontent.com/53658271/120931270-6e313c00-c6b6-11eb-8cca-d3f12be9de97.png)

**Reply by sezanzeb** [reply]
what about `m(Shift_L, h(comma))` instead of `k(ö)` as mentioned in my first comment?

**Reply by RonSMeyer** [reply]
Now that's not going to type the "ö" character on the screen, is it. Never mind. You don't get it. This software is not going to work. Forget it.

**Reply by sezanzeb** [reply]
Thanks for the somewhat-rude answer. I happen to have a full-time job, am busy with moving to a new place and still try to take the time to solve your issues with it timely.

Try this: https://github.com/sezanzeb/key-mapper/blob/main/readme/usage.md#how-to-use-unavailable-symbols

---

I thought that 'ö' is already available in your xmodmap and that you try to map the combination of that key to your keyboard as a single key-press. Sorry for that, I guess it's not present in your xmodmap.

> additionally Alt-L+keycode47 types the character "ö"

I understood that as a description of the givens instead of a description of what you desire to do.
